### DISCUSSION TITLE
[Bug] Multiple applications  work with satellite , service instances not correct

### DISCUSSION THREAD

**Discussion by guangdashao** [apache/skywalking#9219] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Satellite (apache/skywalking-satellite)

### What happened

oap server: 9.0.0
satellite : 1.0.0
java agent: 8.10.0

Multiple applications  use their own agents to send data into the same satellite :
application a , instance id is : 10.151.31.75,10.151.31.76
application b, instance id is : 10.60.41.41
application c, instance  id is : 10.151.31.122
satellite work in ip: 10.151.18.210

java agent config like:
-javaagent:/app/skywalking/skywalking-agent/skywalking-agent.jar=agent.service_name=invest::backend,agent.namespace=invest,agent.cluster=invest,agent.instance_name='10.60.41.41',logging.file_name='invest-skywalking-api.log'

then , instances 10.151.31.75,10.151.31.76,10.60.41.41,10.151.31.122 will in the three applications instance page,like this:
![image](https://user-images.githubusercontent.com/54162228/173519531-ae5846d5-6fa4-4a35-a26d-3c41989972c1.png)

![image](https://user-images.githubusercontent.com/54162228/173519817-7c2f86b8-0799-4bc6-9960-6e3400e2b5dd.png)

### What you expected to happen

The instance appears under the correct service

### How to reproduce

oap server: 9.0.0
satellite : 1.0.0
java agent: 8.10.0

Multiple applications  use their own agents to send data into the same satellite :
application a , instance id is : 10.151.31.75,10.151.31.76
application b, instance id is : 10.60.41.41
application c, instance  id is : 10.151.31.122
satellite work in ip: 10.151.18.210

java agent config like:
-javaagent:/app/skywalking/skywalking-agent/skywalking-agent.jar=agent.service_name=invest::backend,agent.namespace=invest,agent.cluster=invest,agent.instance_name='10.60.41.41',logging.file_name='invest-skywalking-api.log'

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
@mrproliu Do you have any suggestion/clue about this?

**Comment by wu-sheng** [comment]
Instance ID is combined through service name and instance name. Are you reporting through different services?

**Reply by wu-sheng** [reply]
https://github.com/apache/skywalking/blob/master/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/IDManager.java#L105-L114

This is how ID works.

**Reply by wu-sheng** [reply]
You should know that one instance's owner(service) can't be updated to belong to multiple services. 
So, what you described may not make sense.

**Reply by guangdashao** [reply]
i report different services with different service name and instance name : like
`
 -javaagent:/app/skywalking/skywalking-agent/skywalking-agent.jar=agent.service_name=admin,agent.instance_name='admin@10.151.18.84',logging.file_name='admin-skywalking-api.log'
 -javaagent:/app/skywalking/skywalking-agent/skywalking-agent.jar=agent.service_name=oauth,agent.instance_name='oauth@10.151.18.84',logging.file_name='admin-skywalking-api.log'

 -javaagent:/app/skywalking/skywalking-agent/skywalking-agent.jar=agent.service_name=admin,agent.instance_name='admin@10.151.18.216',logging.file_name='admin-skywalking-api.log'
 -javaagent:/app/skywalking/skywalking-agent/skywalking-agent.jar=agent.service_name=oauth,agent.instance_name='oauth@10.151.18.216',logging.file_name='admin-skywalking-api.log'
`
the agent all collect to the same satellite。

then ,instance oauth@10.151.18.216,oauth@10.151.18.84 will in the service admin； like

![image](https://user-images.githubusercontent.com/54162228/173718824-c1792423-f5ba-4a94-a8a4-5fc5729f3198.png)

![image](https://user-images.githubusercontent.com/54162228/173718762-09868ae8-61c9-4617-ba72-5b9f74a52046.png)



should i set up a separate satellite for each service，satellite a for admin ,satellite b for gateway?

**Reply by wu-sheng** [reply]
> should i set up a separate satellite for each service，satellite a for admin ,satellite b for gateway?

No, you don't need to.

@mrproliu Could you take a look?

**Reply by mrproliu** [reply]
Could you please provide more information on that? 
1. Which kind of traffics are transmitted from the agent to OAP? Such as trace, meter or other protocol data, you could see it on the satellite logs. 
2. If you just send the data to the OAP without satellite, is the data correct?

**Reply by guangdashao** [reply]
> Could you please provide more information on that?
> 
> 1. Which kind of traffics are transmitted from the agent to OAP? Such as trace, meter or other protocol data, you could see it on the satellite logs.
> 2. If you just send the data to the OAP without satellite, is the data correct?

1、 There have traffic. if the instance name is correct ,
![image](https://user-images.githubusercontent.com/54162228/173723283-e8987341-c6b6-4292-bb54-657c3b1245f0.png)
when instance name is wrong， there no traffic
![image](https://user-images.githubusercontent.com/54162228/173723345-773c7096-831a-4b8d-a7df-6117d29b1c66.png)

2、I haven't tested using agent directly，but i use Nginx LUA Agent ， the instance data is correct。

**Reply by mrproliu** [reply]
> when instance name is wrong， there no traffic.

Is all the data missing? Or part of data? For example, the JVM, Trace has data when the service and instance name is not match. 

> I haven't tested using agent directly，but i use Nginx LUA Agent ， the instance data is correct。

If conditions permit, it is recommended to try in Java Agent, because there are richer data protocols.

**Reply by wu-sheng** [reply]
A tip, showcase, https://skywalking.apache.org/docs/skywalking-showcase/latest/readme/, is using this satellite version with multiple instances/services. Nothing mixed up,  you could cross check with your env

**Reply by guangdashao** [reply]
if i stop oap servver ,and delete indexes in es, then i start oap server , the traffic  transmitted from the agent to OAP with the satellite are correct ,but if i set up a new server to connect satellite ,all data will be wrong.

**Reply by guangdashao** [reply]
> > when instance name is wrong， there no traffic.
> 
> Is all the data missing? Or part of data? For example, the JVM, Trace has data when the service and instance name is not match.

i have found logs data is not match。a service's log will in b service's logs page。

> 
> > I haven't tested using agent directly，but i use Nginx LUA Agent ， the instance data is correct。
> 
> If conditions permit, it is recommended to try in Java Agent, because there are richer data protocols.

i  try in Java Agent , instance data are correct.
![image](https://user-images.githubusercontent.com/54162228/173725567-222a6a3d-dcdd-499a-bb14-bca4b525fae7.png)

**Reply by mrproliu** [reply]
> but if i set up a new server to connect satellite ,all data will be wrong.

You mean when a new application connect to satellite then all data are wrong? It's not make sense. What's difference between the new application and old application?

**Reply by guangdashao** [reply]
> > but if i set up a new server to connect satellite ,all data will be wrong.
> 
> You mean when a new application connect to satellite then all data are wrong? It's not make sense. What's difference between the new application and old application?

new application and old application is not the same application。

for example ,i have an application A ,this appication A has two instances: A-1,A-2，and  the agent of instances: A-1,A-2 connect to satellite S. at first ,all things is right。
but ,when i start applicatin B , B has one instances ,B-1,and the agent of B-1 also connect to satellite S。
the instance B-1 will in  appication A , and maybe instances: A-1,A-2 will in applicatin B too.

**Reply by guangdashao** [reply]
work with satellite , where restart all service , logs are wrong .like this:

![image](https://user-images.githubusercontent.com/54162228/173735229-2dc58d56-7f1a-4a75-93a4-2652926b9ca4.png)
![image](https://user-images.githubusercontent.com/54162228/173735395-dbb645cf-1de9-4718-8bcd-00d890a3cffe.png)

**Reply by mrproliu** [reply]
Yes, we found that log transmission through satellite has a bug. I will create another issue to track this. Thanks!
